types/lru-cache/lru-cache-tests.ts

```diff
@@ -10,9 +10,9 @@ const foo = {
     foo() {}
 };
 
-const cache = LRU<string, Foo>();
-cache; // $ExpectType Cache<string, Foo>
-LRU<string, Foo>({ // $ExpectType Cache<string, Foo>
+const cache = new LRUCache<string, Foo>();
```

Comment:
Generally speaking, you can't instantiate a symbol imported with `import *`.  Would it be more appropriate to retain the `import = require` form?

Reply:
> Generally speaking, you can't instantiate a symbol imported with `import *`

Could you provide more information why this is impossible? Module exports class `LRUCache` as

```ts
class LRUCache {}

module.exports = LRUCache;
```

Therefore, it can be imported as `import *`.

Reply:
This is a coincidence that it actually works. It wouldn't work any more if you'd remove `declare namespace LRUCache ...` declaration from the declaration file.

You can't actually remove this declaration becase we want to expose interfaces from the module and declaring them in a namespace is the only way to expose them from a module with an `export = ...` export. So this is a necessary evil.

What @amcasey tries to say is that it probably won't work in a native ES module environment, it only works in emulated environments that use CJS-style modules under the hood. You can find more materials on this topic [here](https://github.com/DefinitelyTyped/DefinitelyTyped#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports).

I would also prefer if you change it back to `const LRUCache = require('lru-cache');`, it is simply the only correct solution.

Reply:
Thanks for explanation. This change was reverted.